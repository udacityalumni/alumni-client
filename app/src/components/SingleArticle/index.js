import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Headline from 'grommet-udacity/components/Headline';
import Section from 'grommet-udacity/components/Section';
import Box from 'grommet-udacity/components/Box';
import Article from 'grommet-udacity/components/Article';
import Image from 'grommet-udacity/components/Image';
import Paragraph from 'grommet-udacity/components/Paragraph';
import { Author, ArticleMeta } from 'components';

const SingleArticle = ({
  article,
}) => (
  <Box>
    <Section align="center">
      <Article align="center" className={styles.singleArticle}>
        <Headline align="center">
          {article.title}
        </Headline>
        <Image src={article.feature_image} full="true" />
        <Box
          align="center"
          justify="center"
          size="large"
          pad={{ horizontal: 'small' }}
        >
          <Paragraph>
            {article.content}
          </Paragraph>
        </Box>
      </Article>
    </Section>
    <Section align="center">
      <Author author={article.user} />
    </Section>
    <Section align="center">
      <ArticleMeta article={article} />
    </Section>
  </Box>
);

SingleArticle.propTypes = {
  article: PropTypes.object.isRequired,
};

export default cssModules(SingleArticle, styles);
