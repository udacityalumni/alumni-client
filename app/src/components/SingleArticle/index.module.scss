.heading {
  padding: 20px;
}

.singleArticle {
  position: relative;
  width: 100%;
  background: #fff;
  box-shadow: 0 2px 4px 0 rgba(167, 157, 144, 0.52);
  max-width: 100vw;
  box-sizing: border-box;
  padding: 15px 10px !important;
}
