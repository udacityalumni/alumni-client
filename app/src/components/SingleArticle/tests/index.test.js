import SingleArticle from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
/* eslint-disable */
describe('<SingleArticle />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <SingleArticle
        article={{"id":1,"title":"Creditable coverage Hcfa common procedure coding system Group health plan","featured":false,"spotlighted":true,"content":"Care plan eligible dependent free-look period waiting period incurral date board certified health care provider HCPCS. Agent of record PCP limited policy morbidity medigap health savings account credentialing medical necessity open enrollment. Creditable coverage agent of record renewal disenroll IPA medicare. Accumulation period contract year policy benefit cap medicare supplement creditable coverage.","feature_image":"https://robohash.org/magnietnon.png?size=300x300","status":"published","slug":"creditable-coverage-hcfa-common-procedure-coding-system-group-health-plan","user":{"id":3,"email":"andreas@gmail.com","created_at":"2016-09-30T02:39:57.297Z","updated_at":"2016-10-03T18:49:42.965Z","name":"Andreas Daimainger","avatar":"https://avatars3.githubusercontent.com/u/13679375?v=3&s=466","role":"user"},"created_at":"2016-09-30T02:39:57.333Z","updated_at":"2016-09-30T02:42:00.487Z"}}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
