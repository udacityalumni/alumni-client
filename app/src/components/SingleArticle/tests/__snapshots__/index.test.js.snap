exports[`<SingleArticle /> should render with default props 1`] = `
<Box
  announce={false}
  direction="column"
  focusable={true}
  pad="none"
  responsive={true}
  tag="div">
  <Section
    align="center"
    pad={
      Object {
        "vertical": "medium"
      }
    }>
    <undefined
      article={
        Object {
          "content": "Care plan eligible dependent free-look period waiting period incurral date board certified health care provider HCPCS. Agent of record PCP limited policy morbidity medigap health savings account credentialing medical necessity open enrollment. Creditable coverage agent of record renewal disenroll IPA medicare. Accumulation period contract year policy benefit cap medicare supplement creditable coverage.",
          "created_at": "2016-09-30T02:39:57.333Z",
          "feature_image": "https://robohash.org/magnietnon.png?size=300x300",
          "featured": false,
          "id": 1,
          "slug": "creditable-coverage-hcfa-common-procedure-coding-system-group-health-plan",
          "spotlighted": true,
          "status": "published",
          "title": "Creditable coverage Hcfa common procedure coding system Group health plan",
          "updated_at": "2016-09-30T02:42:00.487Z",
          "user": Object {
            "avatar": "https://avatars3.githubusercontent.com/u/13679375?v=3&s=466",
            "created_at": "2016-09-30T02:39:57.297Z",
            "email": "andreas@gmail.com",
            "id": 3,
            "name": "Andreas Daimainger",
            "role": "user",
            "updated_at": "2016-10-03T18:49:42.965Z"
          }
        }
      } />
  </Section>
  <Section
    align="center"
    pad={
      Object {
        "vertical": "medium"
      }
    }>
    <undefined
      author={
        Object {
          "avatar": "https://avatars3.githubusercontent.com/u/13679375?v=3&s=466",
          "created_at": "2016-09-30T02:39:57.297Z",
          "email": "andreas@gmail.com",
          "id": 3,
          "name": "Andreas Daimainger",
          "role": "user",
          "updated_at": "2016-10-03T18:49:42.965Z"
        }
      } />
  </Section>
  <Section
    align="center"
    pad={
      Object {
        "vertical": "medium"
      }
    }>
    <undefined
      article={
        Object {
          "content": "Care plan eligible dependent free-look period waiting period incurral date board certified health care provider HCPCS. Agent of record PCP limited policy morbidity medigap health savings account credentialing medical necessity open enrollment. Creditable coverage agent of record renewal disenroll IPA medicare. Accumulation period contract year policy benefit cap medicare supplement creditable coverage.",
          "created_at": "2016-09-30T02:39:57.333Z",
          "feature_image": "https://robohash.org/magnietnon.png?size=300x300",
          "featured": false,
          "id": 1,
          "slug": "creditable-coverage-hcfa-common-procedure-coding-system-group-health-plan",
          "spotlighted": true,
          "status": "published",
          "title": "Creditable coverage Hcfa common procedure coding system Group health plan",
          "updated_at": "2016-09-30T02:42:00.487Z",
          "user": Object {
            "avatar": "https://avatars3.githubusercontent.com/u/13679375?v=3&s=466",
            "created_at": "2016-09-30T02:39:57.297Z",
            "email": "andreas@gmail.com",
            "id": 3,
            "name": "Andreas Daimainger",
            "role": "user",
            "updated_at": "2016-10-03T18:49:42.965Z"
          }
        }
      } />
  </Section>
</Box>
`;
